src/libsyncengine/jobs/network/API_v2/upload_session/driveuploadsession.cpp

```diff
@@ -109,4 +111,26 @@ bool DriveUploadSession::handleCancelJobResult(const std::shared_ptr<UploadSessi
 
     return true;
 }
+
+void DriveUploadSession::abort() {
+    AbstractUploadSession::abort();
+    setVfsForceStatusCallback(nullptr);
+
+#ifdef __APPLE__
+    // Removing all extended attributes so that the file that failed to be uploaded is not identified as a dehydrated placeholder
+    // by the application when restarting the synchronization.
+
+    const SyncPath &localPath = getFilePath();
+    static const std::vector<const char *> extendedAttributes = {EXT_ATTR_STATUS, EXT_ATTR_PIN_STATE};
+
+    for (const auto attribute: extendedAttributes) {
```

Comment:
**Questions.**
- Should we remove all attributes or just `EXT_ATTR_STATUS`? (What is the definition of a placeholder for the application?)
- Is a missing permission an actual problem for the user?
- In case of unexpected failure, should the exit code be changed?

Reply:
A file is a placeholder if it has the `EXT_ATTR_STATUS` extended attribute (see `LiteSyncExtConnector::vfsGetStatus`). So it should be enough to remove just this one. However, there is no point to keep only the pin state.
Moreover, the definition of a placeholder based solely on the status is also arguable. Perhaps basing it on the pinstate instead could help us display the correct status in Finder without risking to allow the deshydration of a not yet synced item.

Reply:
> In case of unexpected failure, should the exit code be changed?

Maybe again ^^ but what can we do if the access rights have been removed 🥲